cluster/gce/upgrade.sh

```diff
@@ -439,6 +439,75 @@ function do-node-upgrade() {
   echo "== Finished upgrading nodes to ${KUBE_VERSION}. ==" >&2
 }
 
+CURRENT_COREDNS_VERSION=$(${KUBE_ROOT}/cluster/kubectl.sh -n kube-system get deployment coredns -o=jsonpath='{$.spec.template.spec.containers[:1].image}' | cut -d ":" -f 2)
+
+function update-coredns-config() {
+  # Get the current CoreDNS version
+  local -r coredns_addon_path="/etc/kubernetes/addons/0-dns/coredns"
+  local -r tmpdir=/tmp
+  local -r download_dir=$(mktemp --tmpdir=${tmpdir} -d coredns-migration.XXXXXXXXXX)
+
+  # Get the new installed CoreDNS version
+  sleep 10
+  NEW_COREDNS_VERSION=$(${KUBE_ROOT}/cluster/kubectl.sh -n kube-system get deployment coredns -o=jsonpath='{$.spec.template.spec.containers[:1].image}' | cut -d ":" -f 2)
```

Comment:
Humm.. It seems like we only wait for kube-apiserver to respond before calling `update-coredns-config()`. It may take a bit longer for CoreDNS deployment to be updated in practice. Anyway that we can know for sure the deployment is the updated one?

Reply:
I am not sure of a way to know this. It's the reason I've added `sleep` before fetching the version.

Reply:
We can possibly retrieve the CoreDNS version (or `resourceVersion`, etc.) before and after the upgrade, and make sure it was changed?

Reply:
Thank you, I've added the check now.